### ISSUE TITLE
Feature request: EZoom keyboard (caret) tracking

### ISSUE THREAD

**Issue by daniahl** [compiz-reloaded/compiz-plugins-main#2]
As a visually impaired user, I depend on having full-screen zoom on all the time. The EZoom plugin does this beautifully, but there is one missing feature which would make it amazing - keyboard (caret) tracking. This is important e.g. when writing in a document, where at present the zoomed-in region won't move to follow the caret, thus you have to reach for the mouse all the time to see what text you are entering.

I am trying to switch to Linux, and this is really the only big feature I am missing, accessibility-wise.

**Comment by soreau** [comment]
Hi daniahl, this feature has been requested before, see https://bugs.launchpad.net/ubuntu/+source/compiz/+bug/727290 for instance. The trick is to be able to get the coordinates of the text cursor in the window. There is at-spi but I'm not sure if it provides access to what's needed for this feature.

**Comment by daniahl** [comment]
Thanks for the reply!
I do think at-spi would have that information, since Orca relies on that if I'm not mistaken. I do hope someone is looking into this. I'll check out the other bug report.

**Comment by daniahl** [comment]
It looks like someone has got this working through AT-SPI: https://github.com/gloob/compiz-tracking

Would it be possible to use the Accessibility and modified EZoom plugins from that repo in compiz-reloaded?

**Comment by soreau** [comment]
Yes, the code could be backported but I'm having trouble finding a specific patch detailing what all needs to happen. I'm not sure if other changes are required to core additionally. It appears the commits are spread throughout the branch you referenced.

**Comment by Equipe-ADVL** [comment]
Hello,

Hello, we have developed a PYTHON script for this: Tracking FOCUS...
Equipe-advl
